
R version 4.3.0 (2023-04-21) -- "Already Tomorrow"
Copyright (C) 2023 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> library(testthat)
> library(pkgdown)
> 
> test_check("pkgdown")
[ FAIL 1 | WARN 0 | SKIP 85 | PASS 560 ]

══ Skipped tests ═══════════════════════════════════════════════════════════════
• dir_exists(path)[[1]] is not TRUE (1)
• dir_exists(test_path("assets/site-dot-github/.github")) is not TRUE (2)
• empty test (1)
• On CRAN (81)

══ Failed tests ════════════════════════════════════════════════════════════════
── Error ('test-build-articles.R:73:3'): html widgets get needed css/js ────────
Error in `render_rmarkdown(pkg, input = input, output = output_file, output_format = format, 
    output_options = options, quiet = quiet)`: Failed to render RMarkdown
Caused by error:
! in callr subprocess.
Caused by error:
! there is no package called 'diffviewer'
Backtrace:
     ▆
  1. ├─testthat::expect_output(path <- build_article("widget", pkg)) at test-build-articles.R:73:2
  2. │ └─testthat:::quasi_capture(...)
  3. │   ├─testthat (local) .capture(...)
  4. │   │ └─testthat::capture_output_lines(code, print, width = width)
  5. │   │   └─testthat:::eval_with_output(code, print = print, width = width)
  6. │   │     ├─withr::with_output_sink(path, withVisible(code))
  7. │   │     │ └─base::force(code)
  8. │   │     └─base::withVisible(code)
  9. │   └─rlang::eval_bare(quo_get_expr(.quo), quo_get_env(.quo))
 10. └─pkgdown::build_article("widget", pkg)
 11.   └─pkgdown:::render_rmarkdown(...)
 12.     └─base::tryCatch(...)
 13.       └─base (local) tryCatchList(expr, classes, parentenv, handlers)
 14.         └─base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
 15.           └─value[[3L]](cond)
 16.             └─rlang::abort("Failed to render RMarkdown", parent = cnd)

[ FAIL 1 | WARN 0 | SKIP 85 | PASS 560 ]
Error: Test failures
Execution halted
